core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -243,16 +243,27 @@ class RequestSendThread(val controllerId: Int,
                         val stateChangeLogger: StateChangeLogger,
                         name: String,
                         val controllerChannelManager: ControllerChannelManager)
-  extends ShutdownableThread(name = name) {
+extends ShutdownableThread(name = name) with KafkaMetricsGroup {
 
   logIdent = s"[RequestSendThread controllerId=$controllerId] "
 
+  private val QueueTimeMetricName = "QueueTime"
+
   private val socketTimeoutMs = config.controllerSocketTimeoutMs
 
   private val controllerRequestMerger = new ControllerRequestMerger
 
   private var firstUpdateMetadataWithPartitionsSent = false
 
+  private var requestEnqueueTimeMs = time.milliseconds()
```

Comment:
Suggesting a more verbose name for this variable `inflightRequestEnequeueTimeMs`
Also since this variable is being accessed from multiple threads, it needs to be volatile.